### PR TITLE
Remove old processor detection code on Z

### PR DESCRIPTION
**Pull request by harryyu1994**
#### Summary of changes:
- Retire old version cpu detection code on Z
- Reasons for the removal: 
  - We kept the old version because we wanted to make sure new version and old version give us the same behaviour
  - Now it's been a few month and we don't see any bugs with the new version there's no reason to keep the old version
  - Also we want to prevent developers from using the old version.

#### Old design
- Initializes the `TR_Processor` struct via j9port library
  - Initialization occurs in `TR_J9VM::initializeProcessorType()`
  - obtain cpu information using `CPU::TO_PORTLIB_getJ9ProcessorDesc()`
  - `TR::Compiler->target.cpu.setProcessor` to set processor 
- Fields and methods associated with the old design live in `Class CPU`
- Common APIs:
  - `CPU::TO_PORTLIB_getJ9ProcessorDesc()`
  - `j9sysinfo_processor_has_feature(processorDesc, feature)` (j9port library API)
  - `CPU::id()`
  - `CPU::setProcessor(TR_Processor)`

#### New design
- Initializes the `OMRProcessorDesc` struct via omrport library
  - `CPU::detect()` to initialize `OMRProcessorDesc processorDescription`
  - `CPU::applyUserOptions()` to apply any debug options after initialization
  - `TR::Compiler->target.cpu` singleton variable that holds the CPU instance
  - `comp()->target().cpu` per compilation variable which is used for cross-cpu compilation support
    - For JIT we initialize `comp()->target().cpu` with `TR::Compiler->target.cpu` (Host CPU)
    - For AOT we initialize `comp()->target().cpu` with `TR::Compiler->relocatableTarget.cpu` (Portable CPU)
- Fields and methods associated with the new design live in `Class CPU`
- Common APIs:
  - `CPU::supportsFeature(feature)`
  - `CPU::isAtLeast(OMRProcessorArchitecture)`
- Benefits:
  - Common to all 3 platforms
  - Simpler interface
  - Cross-CPU compilation support (Use one cpu for AOT and another for JIT)

Depends on: https://github.com/eclipse/omr/pull/5541
Issue: #10578 

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

File: runtime/compiler/env/ProcessorDetection.cpp [hunk 1]
```diff
@@ -653,62 +653,12 @@ TR_J9VM::initializeProcessorType()
    if (TR::Compiler->target.cpu.isZ())
       {
 #if defined(TR_HOST_S390)
-      initializeS390ProcessorFeatures();
-
 #if defined(J9ZOS390)
       // Cache whether current process is running in Supervisor State (i.e. Control Region of WAS).
```
Comment (fjeremic) [inline runtime/compiler/env/ProcessorDetection.cpp:657]: So where are things initialized now? Is it in OMR?

Comment (harryyu1994) [reply runtime/compiler/env/ProcessorDetection.cpp:657]: Step 1. OMR detects and initializes the CPU using omr port library.

```cpp
void
OMR::CompilerEnv::initializeHostEnvironment()
   {

   // Host processor bitness
   //
#ifdef TR_HOST_64BIT
   host.setBitness(TR::bits_64);
#elif TR_HOST_32BIT
   host.setBitness(TR::bits_32);
#else
   host.setBitness(TR::bits_unknown);
#endif

   // Initialize the host CPU by querying the host processor
   //
   host.cpu = TR::CPU::detect(TR::Compiler->omrPortLib);
```

Step 2. OpenJ9 calls `J9::Z::CPU::applyUserOptions()` to apply command line options from user and initialize the J9-specific features that couldn't be done in OMR.

```cpp
void
TR_J9VM::initializeProcessorType()
   {
   TR_ASSERT(_compInfo,"compInfo not defined");
   TR::Compiler->target.cpu.applyUserOptions();
```

Comment (harryyu1994) [reply runtime/compiler/env/ProcessorDetection.cpp:657]: `applyUserOptions()` mirrors the things that were done in `initializeS390ProcessorFeatures()`

Comment (fjeremic) [reply runtime/compiler/env/ProcessorDetection.cpp:657]: Thanks for the explanation! We'll wait until the OMR change has propagated.

**Comment by fjeremic** [comment]
Jenkins test sanity zlinux jdk11

**Comment by harryyu1994** [comment]
There's a small portion of common code. May want to test power and x86 as well.

**Comment by fjeremic** [comment]
enkins test sanity all jdk8

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8

**Review by fjeremic — APPROVED** [review]

**Pull request merged**